File: modules/expo-backup-exclusions/android/src/main/java/expo/modules/backupexclusions/ExpoBackupExclusionsModule.kt
package expo.modules.backupexclusions

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import expo.modules.kotlin.Promise

/**
 * Android backup exclusion is handled at build time via fullBackupContent and
 * dataExtractionRules XML. This module exists so the app can call excludeFromBackup()
 * on both platforms; on Android it is a no-op.
 */
class ExpoBackupExclusionsModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("ExpoBackupExclusions")

        AsyncFunction("excludeFromBackup") { promise: Promise ->
            // No-op on Android; backup exclusions are configured via XML at build time
            promise.resolve(null)
        }
    }
}

File: modules/expo-backup-exclusions/src/ExpoBackupExclusionsModule.ts
import { requireNativeModule } from 'expo-modules-core';

let module: { excludeFromBackup: () => Promise<void> };

try {
  module = requireNativeModule('ExpoBackupExclusions');
} catch {
  console.warn(
    '[expo-backup-exclusions] Native module not found. ' +
      'Run `npx expo run:ios` or `npx expo run:android` to rebuild with the native module.'
  );

  module = {
    excludeFromBackup: () => Promise.resolve(),
  };
}

export default module;
